app/services/articles/analytics_updater.rb

```diff
@@ -0,0 +1,57 @@
+module Articles
+  class AnalyticsUpdater
+    def initialize(context = "default")
```

Comment:
I don't understand what this context is for. The initializer also has a different arg list then user, which means that when you call `analytics_updater.call(user)` in the job, `@context` will be assigned to user and that call will fail.

I don't think you're testing for this in the tests, because they all use the instance, while in the job you're using the class method.

This method should take a user and store it as `@user = user` and then the call method should use that

Reply:
Oups, corrected.
Regarding the context, this is existing code, and we might need it in the future, So I left it.